sound/soc/sof/intel/pci-tgl.c

```diff
@@ -33,18 +33,18 @@ static const struct sof_dev_desc tgl_desc = {
 	.dspless_mode_supported	= true,		/* Only supported for HDaudio */
 	.default_fw_path = {
 		[SOF_IPC_TYPE_3] = "intel/sof",
-		[SOF_IPC_TYPE_4] = "intel/avs/tgl",
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4/tgl",
```

Comment:
You're missing two things
a) an explanation on the (non)-impact of this patch outside the SOF development community. I would bet Marc Brown and Takashi Iwai would be concerned about breaking existing user setups
b) a change of topology names for MTL/LNL if you want to be consistent and lose the avs/ace split.
c) a clear plan with CI folks, if we merge this then everything is broken for multiple days. We can't have this happen.

c) is the most important part, until we have passing tests this cannot be merged.

Reply:
for (b), it's too late to do, we are in final stages of v2.7 of release and that will install MTL tplgs to sof-ace-tplg. so kernel has to support this layout and we have to do the migration to intel/sof-ipc4-tplg in stages.

Reply:
> You're missing two things
> a) an explanation on the (non)-impact of this patch outside the SOF development community. I would bet Marc Brown and Takashi Iwai would be concerned about breaking existing user setups

I agree, this needs to be spelled out clearly

> b) a change of topology names for MTL/LNL if you want to be consistent and lose the avs/ace split.

I have a draft sof-docs PR to cover the planned naming: https://github.com/thesofproject/sof-docs/pull/476

But changing MTL/LNL has is a tricky one as we are just about to release 2023.09 (also known as v2.7) and that is set to use the `sof-ace-tplg` path.

> c) a clear plan with CI folks, if we merge this then everything is broken for multiple days. We can't have this happen.

Right, I think this is a multilayer issue and I would add **sof-bin** into the mix.
So far I think jenkins, deploy needs changes to work with the new paths, I'm not sure about the sof-test and probably a change in SOF itself might be desired.

> c) is the most important part, until we have passing tests this cannot be merged.

I think the volatile part is the tplg, CI is specifying the path and firmware file name via module parameter, never uses the default path. I'm not sure about the tplg, but if it is using the default path then a symlink is needed. We cannot synchronize kernel and infra PRs to make sure all is good.
My (now closed) PR would have been able to handle this under the hood: https://github.com/thesofproject/linux/pull/4603

Reply:
> c) a clear plan with CI folks, if we merge this then everything is broken for multiple days. We can't have this happen.

Not multiple days, one day should be enough if it's well tested ahead of time. We've seen worse downtimes for worse reasons.

Please IGNORE CI in this equation. CI is a means to an end, CI is here to SERVE developers (= xtensa-build-zephyr.py output) and customers (sof-bin), not the other way round. Also, CI is currently full of firmware deployment secrets and this is a good opportunity to simplify and get rid of a few of these secrets.

If everything outside CI is simple, clean, consistent and well documented in sof-docs, then CI will be welcome that breath of fresh air and simplicity and will be delighted to cleanup and simplify ancient and arcane deployment secrets.